@hts-forking/src/slotmap.js

```diff
@@ -0,0 +1,175 @@
+/*-
+ * Hedera Hardhat Forking Plugin
+ *
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { strict: assert } = require('assert');
+const { keccak256 } = require('ethers');
+
+const { toIntHex256, toSnakeCase } = require('./utils');
+const {
+    storageLayout: { storage, types },
+} = require('../resources/HtsSystemContract.json');
+
+/** @typedef {string | ((mirrorNode: Pick<import('.').IMirrorNodeClient, 'getAccount'>, blockNumber: number) => Promise<string>)} Value */
+
+class SlotMap {
```

Comment:
Is this class logic called out in a doc or current README or in process to document.
I think it's really powerful.
At the least you should update the PR description a bit more